src/app/services/sync.service.ts

```diff
@@ -183,12 +182,10 @@ export class SyncService {
 		this.syncAllSettings()
 	}
 
-	async deleteAllTemp() {
+	async deleteAllSyncChangeKeys() {
 		const allNotifyKeys = await this.getAllSyncChangeKeys()
 		for (const key of allNotifyKeys) {
-			if (key.indexOf('temponly') >= 0) {
-				this.storage.remove(key)
-			}
+			this.storage.remove(key)
```

Comment:
@manuelsc, 'temponly' was never set so the sync change keys have never been cleaned up properly.
This should be cleaner now, please pay special attention to this.

Reply:
You can delete the `deleteAllSyncChangeKeys` method and its invocation. `temponly` seems to be a relict from the past for a special case which is not in the codebase anymore. SyncChangeKeys should never be deleted since they store information of the last setting change.

Reply:
The way it was before, the method did not do anything as `temponly` wasn't used anymore, right. I have changed that so it deletes all "sync change keys" once they are synced and to me that makes perfect sense.

Consider the following scenario:
1) User opens notifications, changes the "validator slashed" toggle and leaves the notifications.
2) User opens notifications again, changes nothing and leaves the notifications.

Without `deleteAllSyncChangeKeys`, the following would happen:
1) Once the user toggles "validator slashed", a `NOTIFY_SYNCCHANGE` for "validators slashed" is added via `setChanged`.
2) Once the user leaves the notifications, this gets synced.
3) Once the user re-opens and leaves the notifications without changing anything, the `NOTIFY_SYNCCHANGE` for "validators slashed" is still present and another sync would happen (even though nothing has changed).

WITH `deleteAllSyncChangeKeys` in place, the `NOTIFY_SYNCCHANGE` for "validators slashed" would be deleted once the sync happened. Re-openeing and leaving the notifications without changing anything would not result in another sync.

Isn't this the desired behavior?

Reply:
Once you press a toggle a change object is stored. This object has multiple important data, network, value, threshold, if subscribe or unsubscribe. And they also contain two timestamps, `lastChanged` and `lastSynced`. Sync service retrieves all changes from the storage and checks whether `lastChanged` > `lastSynced` and only then syncs the changes. If a sync has been successful, the object will be updated and `lastSynced` is set to the current time. For more information on how this works take a look at `setLastSynced()` and its invocation as well as `lastSynced` and its references.  
  
As I mentioned, `temponly` is a relic from the past. You can safely delete the method and its usage altogether. The behavior stays unchanged due to `lastSynced`.

Reply:
Many thanks for the detailed explanation. You are absolutely right.
I have made the requested changes.